cdap-app-fabric/src/main/java/co/cask/cdap/test/internal/AppFabricClient.java

```diff
@@ -69,60 +77,88 @@
     .create();
 
   private final AppFabricHttpHandler httpHandler;
-  private final ServiceHttpHandler serviceHttpHandler;
   private final LocationFactory locationFactory;
-
-  public AppFabricClient(AppFabricHttpHandler httpHandler, ServiceHttpHandler serviceHttpHandler,
-                         LocationFactory locationFactory) {
+  private final AppLifecycleHttpHandler appLifecycleHttpHandler;
+  private final ProgramLifecycleHttpHandler programLifecycleHttpHandler;
+  private final NamespaceHttpHandler namespaceHttpHandler;
+  private final NamespaceAdmin namespaceAdmin;
+  private final StreamAdmin streamAdmin;
+  private final StreamMetaStore streamMetaStore;
+
+  @Inject
+  public AppFabricClient(AppFabricHttpHandler httpHandler, LocationFactory locationFactory,
+                         AppLifecycleHttpHandler appLifecycleHttpHandler,
+                         ProgramLifecycleHttpHandler programLifecycleHttpHandler,
+                         NamespaceHttpHandler namespaceHttpHandler,
+                         NamespaceAdmin namespaceAdmin, StreamAdmin streamAdmin,
+                         StreamMetaStore streamMetaStore) {
     this.httpHandler = httpHandler;
-    this.serviceHttpHandler = serviceHttpHandler;
     this.locationFactory = locationFactory;
+    this.appLifecycleHttpHandler = appLifecycleHttpHandler;
+    this.programLifecycleHttpHandler = programLifecycleHttpHandler;
+    this.namespaceHttpHandler = namespaceHttpHandler;
+    this.namespaceAdmin = namespaceAdmin;
+    this.streamAdmin = streamAdmin;
+    this.streamMetaStore = streamMetaStore;
   }
 
-  public void reset() {
+  public void reset() throws Exception {
     MockResponder responder = new MockResponder();
     String uri = String.format("/v2/unrecoverable/reset");
```

Comment:
if we remove the condition on line 114, we can also remove this call to unrecoverable/reset.